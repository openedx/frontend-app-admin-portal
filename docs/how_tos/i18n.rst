####################
React App i18n HOWTO
####################

This document has moved to the frontend-platform repo: https://github.com/openedx/frontend-platform/blob/master/docs/how_tos/i18n.rst
