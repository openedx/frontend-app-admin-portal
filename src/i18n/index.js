import { messages as paragonMessages } from '@edx/paragon';

export default [
  paragonMessages,
];
