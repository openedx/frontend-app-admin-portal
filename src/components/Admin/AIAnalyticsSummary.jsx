import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Button, Card, Stack, Badge, useToggle,
} from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';
import { AutoFixHigh, Groups } from '@edx/paragon/icons';
import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
import useAIAnalyticsSummary from '../AIAnalyticsSummary/data/hooks';

export const SUMMARIZE_ANALYTICS_CLICK_SEGMENT_EVENT_NAME = 'edx.ui.enterprise.admin_portal.summarize_analytics.clicked';
const AnalyticsDetailCard = ({
  onClose,
  isLoading,
  error,
  data,
}) => (
  <Card className="mt-3 mb-4" isLoading={isLoading}>
    <Card.Section>
      <Badge variant="light" className="mb-3 font-weight-semibold">
        <FormattedMessage id="adminPortal.analyticsCardBetaButton" defaultMessage="Beta" />
      </Badge>
      <Stack gap={1} direction="horizontal">
        <p className="card-text text-justify small">
          <FormattedMessage
            id="adminPortal.analyticsCardText"
            defaultMessage={
              error
                ? `An error occurred: ${error.message}`
                : data || 'Analytics not found.'
            }
          />
        </p>
        <Button variant="link" className="mb-4 ml-3" onClick={onClose}>
          <span className="small font-weight-bold text-gray-800">Dismiss</span>
        </Button>
      </Stack>
      <label className="x-small" htmlFor="poweredBylabel">
        <FormattedMessage id="adminPortal.analyticsCardPoweredBylabel" defaultMessage="Powered by OpenAI" />
      </label>
    </Card.Section>
  </Card>
);

AnalyticsDetailCard.propTypes = {
  onClose: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  error: PropTypes.instanceOf(Error),
  data: PropTypes.string,
};

const AIAnalyticsSummary = ({ enterpriseId, insights }) => {
  const [summarizeCardIsOpen, showSummarizeCard, hideSummarizeCard] = useToggle(false);
  const [trackProgressCardIsOpen, showTrackProgressCard, hideTrackProgressCard] = useToggle(false);

  const { data: analyticsSummary, isLoading, error } = useAIAnalyticsSummary(enterpriseId, insights);

  return (
    <>
      <Stack gap={3} direction="horizontal">
        <Button
          variant="outline-primary"
          className="d-sm-inline"
          onClick={() => {
            showSummarizeCard(true);
            hideTrackProgressCard(true);
            sendEnterpriseTrackEvent(enterpriseId, SUMMARIZE_ANALYTICS_CLICK_SEGMENT_EVENT_NAME);
          }}
          data-testid="summarize-analytics"
        >
          <>
            <AutoFixHigh className="mr-2" />
            <FormattedMessage id="adminPortal.summarizeAnalytics" defaultMessage="Summarize Analytics" />
          </>
        </Button>
        {/* Track Progress is currently hidden due to data inconsistency. It will be addressed as part of ENT-7812 */}
        <Button
          variant="outline-primary"
          className="d-none"
          onClick={() => {
            showTrackProgressCard(true);
            hideSummarizeCard(true);
          }}
          data-testid="track-progress"
        >
          <>
            <Groups className="mr-2" />
            <FormattedMessage id="adminPortal.trackProgress" defaultMessage="Track Progress" />
          </>
        </Button>
      </Stack>
      {summarizeCardIsOpen && (
        <AnalyticsDetailCard
          data={analyticsSummary?.learner_engagement}
          onClose={() => hideSummarizeCard(true)}
          isLoading={isLoading}
          error={error}
        />
      )}
      {trackProgressCardIsOpen && (
        <AnalyticsDetailCard
          data={analyticsSummary?.learner_progress}
          onClose={() => hideTrackProgressCard(true)}
          isLoading={isLoading}
          error={error}
        />
      )}
    </>
  );
};

const mapStateToProps = state => ({
  insights: state.dashboardInsights.insights,
});

AIAnalyticsSummary.propTypes = {
  enterpriseId: PropTypes.string.isRequired,
  insights: PropTypes.objectOf(PropTypes.shape),
};

export default connect(mapStateToProps)(AIAnalyticsSummary);
