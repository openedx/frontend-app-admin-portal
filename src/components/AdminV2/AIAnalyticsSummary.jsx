import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Button, Card, Stack, Badge, useToggle,
} from '@openedx/paragon';
import { FormattedMessage, defineMessages } from '@edx/frontend-platform/i18n';
import { AutoFixHigh, Groups } from '@openedx/paragon/icons';
import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
import useAIAnalyticsSummary from '../AIAnalyticsSummary/data/hooks';

export const SUMMARIZE_ANALYTICS_CLICK_SEGMENT_EVENT_NAME = 'edx.ui.enterprise.admin_portal.summarize_analytics.clicked';
const ENABLE_TRACK_PROGRESS = false;

const AnalyticsDetailCard = ({
  onClose,
  isLoading,
  error,
  data,
}) => {
  const messages = defineMessages({
    errorMessage: {
      id: 'adminPortal.analyticsCardText.errorMessage',
      defaultMessage: 'We encountered an issue while fetching analytics data. Kindly try again later or contact support for assistance. (Error: {error_message})',
      description: 'Message shown to the user in case of error returned byt analytics API.',
      values: { error_message: error?.message },
    },
    noContentErrorMessage: {
      id: 'adminPortal.analyticsCardText.noContentErrorMessage',
      defaultMessage: 'Analytics not found.',
      description: 'Message shown to the user in case of empty response returned byt analytics API.',
    },
  });

  return (
    <Card className="mt-3 mb-4" isLoading={isLoading}>
      <Card.Section>
        <Badge variant="light" className="mb-3 font-weight-semibold">
          <FormattedMessage id="adminPortal.analyticsCardBetaButton" defaultMessage="Beta" />
        </Badge>
        <Stack gap={1} direction="horizontal">
          <p className="card-text text-justify small">
            {
              error ? (
                <FormattedMessage {...messages.errorMessage} />
              ) : (
                data || <FormattedMessage {...messages.noContentErrorMessage} />
              )
            }
          </p>
          <Button variant="link" className="mb-4 ml-3" onClick={onClose}>
            <span className="small font-weight-bold text-gray-800">Dismiss</span>
          </Button>
        </Stack>
        <div className="x-small">
          <FormattedMessage id="adminPortal.analyticsCardPoweredBylabel" defaultMessage="Powered by OpenAI" />
        </div>
      </Card.Section>
    </Card>
  );
};

AnalyticsDetailCard.propTypes = {
  onClose: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  error: PropTypes.instanceOf(Error),
  data: PropTypes.string,
};

const AIAnalyticsSummary = ({ enterpriseId, insights, renderOverviewHeading }) => {
  const [summarizeCardIsOpen, showSummarizeCard, hideSummarizeCard] = useToggle(false);
  const [trackProgressCardIsOpen, showTrackProgressCard, hideTrackProgressCard] = useToggle(false);

  const { data: analyticsSummary, isLoading, error } = useAIAnalyticsSummary(enterpriseId, insights);

  return (
    <>
      <Stack gap={3} direction="horizontal" className="justify-content-between">
        {renderOverviewHeading()}

        <Button
          variant="outline-primary"
          className="d-sm-inline"
          onClick={() => {
            showSummarizeCard(true);
            hideTrackProgressCard(true);
            sendEnterpriseTrackEvent(enterpriseId, SUMMARIZE_ANALYTICS_CLICK_SEGMENT_EVENT_NAME);
          }}
          data-testid="summarize-analytics"
        >
          <>
            <AutoFixHigh className="mr-2" />
            <FormattedMessage id="adminPortal.v2.summarizeAnalytics" defaultMessage="Summarize analytics" />
          </>
        </Button>
        {/* Track Progress is currently hidden due to data inconsistency. It will be addressed as part of ENT-7812 */}
        {ENABLE_TRACK_PROGRESS && (
          <Button
            variant="outline-primary"
            className="d-none"
            onClick={() => {
              showTrackProgressCard(true);
              hideSummarizeCard(true);
            }}
            data-testid="track-progress"
          >
            <>
              <Groups className="mr-2" />
              <FormattedMessage id="adminPortal.trackProgress" defaultMessage="Track Progress" />
            </>
          </Button>
        )}
      </Stack>
      {summarizeCardIsOpen && (
        <AnalyticsDetailCard
          data={analyticsSummary?.learner_engagement}
          onClose={() => hideSummarizeCard(true)}
          isLoading={isLoading}
          error={error}
        />
      )}
      {trackProgressCardIsOpen && (
        <AnalyticsDetailCard
          data={analyticsSummary?.learner_progress}
          onClose={() => hideTrackProgressCard(true)}
          isLoading={isLoading}
          error={error}
        />
      )}
    </>
  );
};

const mapStateToProps = state => ({
  insights: state.dashboardInsights.insights,
});

AIAnalyticsSummary.propTypes = {
  enterpriseId: PropTypes.string.isRequired,
  insights: PropTypes.objectOf(PropTypes.shape),
  renderOverviewHeading: PropTypes.func,
};

export default connect(mapStateToProps)(AIAnalyticsSummary);
