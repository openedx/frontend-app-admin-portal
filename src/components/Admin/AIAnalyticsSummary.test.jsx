import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import thunk from 'redux-thunk';
import AIAnalyticsSummary from './AIAnalyticsSummary';

const mockedInsights = {
  learner_progress: {
    enterprise_customer_uuid: 'aac56d39-f38d-4510-8ef9-085cab048ea9',
    enterprise_customer_name: 'Microsoft Corporation',
    active_subscription_plan: true,
    assigned_licenses: 0,
    activated_licenses: 0,
    assigned_licenses_percentage: 0.0,
    activated_licenses_percentage: 0.0,
    active_enrollments: 1026,
    at_risk_enrollment_less_than_one_hour: 26,
    at_risk_enrollment_end_date_soon: 15,
    at_risk_enrollment_dormant: 918,
    created_at: '2023-10-02T03:24:17Z',
  },
  learner_engagement: {
    enterprise_customer_uuid: 'aac56d39-f38d-4510-8ef9-085cab048ea9',
    enterprise_customer_name: 'Microsoft Corporation',
    enrolls: 49,
    enrolls_prior: 45,
    passed: 2,
    passed_prior: 0,
    engage: 67,
    engage_prior: 50,
    hours: 62,
    hours_prior: 49,
    contract_end_date: '2022-06-13T00:00:00Z',
    active_contract: false,
    created_at: '2023-10-02T03:24:40Z',
  },
};
const mockStore = configureMockStore([thunk]);
const store = mockStore({
  portalConfiguration: {
    enterpriseId: 'test-enterprise-id',
  },
  dashboardInsights: mockedInsights,
});

const AIAnalyticsSummaryWrapper = props => (
  <MemoryRouter>
    <Provider store={store}>
      <IntlProvider locale="en">
        <AIAnalyticsSummary
          enterpriseId="test-enterprise-id"
          insights={mockedInsights}
          {...props}
        />,
      </IntlProvider>
    </Provider>
  </MemoryRouter>
);

describe('<AIAnalyticsSummary />', () => {
  it('should render action buttons correctly', () => {
    const tree = renderer
      .create((
        <AIAnalyticsSummaryWrapper
          insights={mockedInsights}
        />
      ))
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should display AnalyticsDetailCard with learner_engagement data when Summarize Analytics button is clicked', () => {
    const wrapper = mount(<AIAnalyticsSummaryWrapper insights={mockedInsights} />);
    wrapper.find('[data-testid="summarize-analytics"]').first().simulate('click');

    const tree = renderer
      .create(<AIAnalyticsSummaryWrapper insights={mockedInsights} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // Currently disabled due to data inconsistencies, will be addressed as a part of ENT-7812.
  // it('should display AnalyticsDetailCard with learner_progress data when Track Progress button is clicked', () => {
  //   const wrapper = mount(<AIAnalyticsSummaryWrapper insights={mockedInsights} />);
  //   wrapper.find('[data-testid="track-progress"]').first().simulate('click');

  //   const tree = renderer
  //     .create(<AIAnalyticsSummaryWrapper insights={mockedInsights} />)
  //     .toJSON();

  //   expect(tree).toMatchSnapshot();
  // });

  it('should handle null analytics data', () => {
    const insightsData = { ...mockedInsights, learner_engagement: null };
    const wrapper = mount(<AIAnalyticsSummaryWrapper insights={insightsData} />);
    wrapper.find('[data-testid="summarize-analytics"]').first().simulate('click');

    const tree = renderer
      .create(<AIAnalyticsSummaryWrapper insights={insightsData} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
