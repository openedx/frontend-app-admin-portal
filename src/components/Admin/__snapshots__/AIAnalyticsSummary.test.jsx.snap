// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AIAnalyticsSummary /> should display AnalyticsDetailCard with learner_engagement data when Summarize Analytics button is clicked 1`] = `
Array [
  <div
    className="pgn__hstack pgn__stack-gap--3"
  >
    <button
      className="d-sm-inline btn btn-outline-primary"
      data-testid="summarize-analytics"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="mr-2"
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 7l.94-2.06L23 4l-2.06-.94L20 1l-.94 2.06L17 4l2.06.94zM8.5 7l.94-2.06L11.5 4l-2.06-.94L8.5 1l-.94 2.06L5.5 4l2.06.94zM20 12.5l-.94 2.06-2.06.94 2.06.94.94 2.06.94-2.06L23 15.5l-2.06-.94zm-1.59-2.67l-4.24-4.24L1.59 18.17l4.24 4.24L18.41 9.83zm-4.2 1.38L12.8 9.8l1.38-1.38 1.41 1.41-1.38 1.38z"
          fill="currentColor"
        />
      </svg>
      Summarize Analytics
    </button>
    <button
      className="d-none btn btn-outline-primary"
      data-testid="track-progress"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="mr-2"
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58A2.01 2.01 0 000 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85A6.95 6.95 0 0020 14c-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
          fill="currentColor"
        />
      </svg>
      Track Progress
    </button>
  </div>,
  ",",
]
`;

exports[`<AIAnalyticsSummary /> should handle null analytics data 1`] = `
Array [
  <div
    className="pgn__hstack pgn__stack-gap--3"
  >
    <button
      className="d-sm-inline btn btn-outline-primary"
      data-testid="summarize-analytics"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="mr-2"
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 7l.94-2.06L23 4l-2.06-.94L20 1l-.94 2.06L17 4l2.06.94zM8.5 7l.94-2.06L11.5 4l-2.06-.94L8.5 1l-.94 2.06L5.5 4l2.06.94zM20 12.5l-.94 2.06-2.06.94 2.06.94.94 2.06.94-2.06L23 15.5l-2.06-.94zm-1.59-2.67l-4.24-4.24L1.59 18.17l4.24 4.24L18.41 9.83zm-4.2 1.38L12.8 9.8l1.38-1.38 1.41 1.41-1.38 1.38z"
          fill="currentColor"
        />
      </svg>
      Summarize Analytics
    </button>
    <button
      className="d-none btn btn-outline-primary"
      data-testid="track-progress"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="mr-2"
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58A2.01 2.01 0 000 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85A6.95 6.95 0 0020 14c-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
          fill="currentColor"
        />
      </svg>
      Track Progress
    </button>
  </div>,
  ",",
]
`;

exports[`<AIAnalyticsSummary /> should render action buttons correctly 1`] = `
Array [
  <div
    className="pgn__hstack pgn__stack-gap--3"
  >
    <button
      className="d-sm-inline btn btn-outline-primary"
      data-testid="summarize-analytics"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="mr-2"
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 7l.94-2.06L23 4l-2.06-.94L20 1l-.94 2.06L17 4l2.06.94zM8.5 7l.94-2.06L11.5 4l-2.06-.94L8.5 1l-.94 2.06L5.5 4l2.06.94zM20 12.5l-.94 2.06-2.06.94 2.06.94.94 2.06.94-2.06L23 15.5l-2.06-.94zm-1.59-2.67l-4.24-4.24L1.59 18.17l4.24 4.24L18.41 9.83zm-4.2 1.38L12.8 9.8l1.38-1.38 1.41 1.41-1.38 1.38z"
          fill="currentColor"
        />
      </svg>
      Summarize Analytics
    </button>
    <button
      className="d-none btn btn-outline-primary"
      data-testid="track-progress"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="mr-2"
        fill="none"
        height={24}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58A2.01 2.01 0 000 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85A6.95 6.95 0 0020 14c-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
          fill="currentColor"
        />
      </svg>
      Track Progress
    </button>
  </div>,
  ",",
]
`;
